include/nbl/video/IDescriptorPool.h

```diff
@@ -29,12 +33,156 @@ class NBL_API IDescriptorPool : public core::IReferenceCounted, public IBackendO
             uint32_t count;
         };
 
-        explicit IDescriptorPool(core::smart_refctd_ptr<const ILogicalDevice>&& dev, uint32_t _maxSets) : IBackendObject(std::move(dev)), m_maxSets(_maxSets) {}
+    private:
+        struct SCombinedImageSampler
+        {
+            core::smart_refctd_ptr<video::IGPUImageView> view;
+            core::smart_refctd_ptr<video::IGPUSampler> sampler;
+        };
+
+        // This construct makes it so that we don't trigger T's constructors and destructors.
+        // TODO(achal): Should we move it to the core namespace?
+        template <typename T>
+        struct alignas(T) StorageTrivializer
+        {
+            uint8_t storage[sizeof(T)];
+        };
+
+    public:
+        explicit IDescriptorPool(core::smart_refctd_ptr<const ILogicalDevice>&& dev, const IDescriptorPool::E_CREATE_FLAGS flags, uint32_t _maxSets, const uint32_t poolSizeCount, const IDescriptorPool::SDescriptorPoolSize* poolSizes) : IBackendObject(std::move(dev)), m_maxSets(_maxSets), m_flags(flags)
+        {
+            memset(m_maxDescriptorCount, 0, asset::EDT_COUNT * sizeof(uint32_t));
```

Comment:
minor stylistic note, `std::fill_n` is the preferred function to use, now

Reply:
Resolved.